atlasdb-commons/src/test/java/com/palantir/util/EclipseCollectionsTest.java

```diff
@@ -32,8 +41,26 @@ class EclipseCollectionsTest {
      * </ul>
      */
     @Test
-    @Timeout(value = 30, unit = TimeUnit.SECONDS) // if timeout is exceeded, class load has likely deadlocked
-    void canLoadClassesWithoutDeadlock() {
-        assertThatCode(EclipseCollections::loadClasses).doesNotThrowAnyException();
+    @Timeout(value = 30, unit = TimeUnit.SECONDS)
+    // if timeout is exceeded, class load has likely deadlocked
+    void canLoadClassesWithoutDeadlock() throws ExecutionException, InterruptedException {
+        ListeningExecutorService executorService = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(4));
+        try {
+            List<ListenableFuture<Boolean>> futures = new ArrayList<>();
+            for (int i = 0; i < 10; i++) {
+                futures.add(executorService.submit(() -> {
+                    assertThatCode(EclipseCollections::loadClasses).doesNotThrowAnyException();
+                    for (String className : EclipseCollections.CLASSES_TO_INITIALIZE) {
+                        assertThat(Class.forName(className)).isNotNull();
```

Comment:
doesn't Class#forName [load the class if it's not already loaded](https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/lang/Class.html#forName(java.lang.String))?